builtins/src/string.rs

```diff
@@ -506,45 +507,52 @@ fn char_is_whitespace(target: SloshChar) -> VMResult<bool> {
     }
 }
 
-///  Usage: (str->int string) -> int
+///  Usage: (->int ?int) -> int
 ///
-///  If string is a valid representation of an integer return that int.  Error if not.
+///  If string or other value is a valid representation of an integer return that int.  Error if not.
 ///
-///  Section: char
+///  Note, if value is a float (or is a string that casts initially to a float) and it is NaN,
+/// +/- infinity, < i56::MIN, or > i56::MAX the function will error.
+///
+///  Section: type
 ///
 ///  Example:
-///  (test::assert-equal 0 (str->int "0"))
-///  (test::assert-equal 101 (str->int "101"))
-///  (test::assert-equal -101 (str->int "-101"))
-///  (test::assert-error (str->int "not int"))
-///  (test::assert-error (str->int "10.0"))
-///  (test::assert-error (str->int "--10"))
-#[sl_sh_fn(fn_name = "str->int")]
-fn str_to_int(target: &str) -> VMResult<i64> {
-    target
-        .parse::<i64>()
-        .map_err(|_e| VMError::new_conversion("Not a valid integer."))
+///  (test::assert-equal 0 (->int "0"))
+///  (test::assert-equal 101 (->int "101"))
+///  (test::assert-equal 1101 (->int :1101))
+///  (test::assert-equal 8 (->int 8.1))
+///  (test::assert-equal -111 (->int "-111"))
+///  (test::assert-equal 10 (->int "10.0"))
+///  (test::assert-error (->int "not int"))
+///  (test::assert-error (->int "--10"))
+/// ;; for some reason this ->sym test case must be last or else there's an error w/ load
+///  (test::assert-equal 1102 (->int (->sym 1102)))
```

Comment:
@sstanfield how do I create a Value::Byte? I want to test that i'm correctly modeling a u8 as a [u8; 7] (least significant bits replaced w/ source u8, everything else left as 0).